standalone/WebSocketExportToOrg.py

```diff
@@ -0,0 +1,141 @@
+** Summary [Optional]
+	 - Host: <host>
+	 - URI: <URI>
+	 - Port: <port>
+	 - Websocket Port: <WebSocket Port>
+	 - Websocket URL: <WebSocket URL>
+	 - Messages Opcodes: 
+	 - Outgoing Messages format: 
+	 - Incoming Messages format: 
+	 - Input Messages encryption: 
+	 - Output Messages encryption: 
+	 - Ping/Pong: 
+	 - Close message: Opcode= , Text= 
+** Switching Protocol
+*** Handshake Request
+	#+BEGIN_SRC html
+	<Http Handshake Request>
+	#+END_SRC
+*** Handshake Respone
+	#+BEGIN_SRC html
+	<Http Handshake Request>
+	#+END_SRC
+** Messages
+*** (#<ChannelID>.<MessageID) <Direction> <Opcode> <Date Time> <Byte Lenght>
+	- Text
+	<Text Payload>
+	- Hex
+	<Hex Payload>
+"""
+
+from org.parosproxy.paros.control import Control
+from org.parosproxy.paros.extension.history import ExtensionHistory
+from org.zaproxy.zap.extension.websocket import ExtensionWebSocket
+from org.zaproxy.zap.extension.websocket import WebSocketChannelDTO
+from org.zaproxy.zap.extension.websocket import WebSocketMessageDTO
+
+import PyOrgMode
+import sys
+import binascii
+
+reload(sys)
+sys.setdefaultencoding('utf-8')
+
+extWebSocket = Control.getSingleton().getExtensionLoader().getExtension(ExtensionWebSocket.NAME)
+extHist = Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.NAME)
+pathToOrgFile = "/path/to/file.org"
+printSummary = True
+base = PyOrgMode.OrgDataStructure()
+channelsList = extWebSocket.getStorage().getTable().getChannelItems()
+
+for channel in channelsList:
+  handshakeMessage = channel.getHandshakeReference().getHttpMessage()
+  # Loop through the Channel List
+  newChannelElement = PyOrgMode.OrgNode.Element()
+  newChannelElement.heading = "(#"+str(channel.id)+"): " + handshakeMessage.getRequestHeader().getHostName() + "\n"
+  newChannelElement.level = 1
+
+  #Print Summary
+  if(printSummary):
+    
+    newSummary = PyOrgMode.OrgNode.Element() 
+    newSummary.heading = "Summary"
+    newSummary.level = 2
+    newSummary.append("\t - Host: " + handshakeMessage.getRequestHeader().getHostName() + "\n")
+    newSummary.append("\t - URI: " + handshakeMessage.getRequestHeader().getMethod() + " " + handshakeMessage.getRequestHeader().getURI().toString() + "\n")
+    newSummary.append("\t - Port: " + str(handshakeMessage.getRequestHeader().getURI().getPort()) + "\n")
+    newSummary.append("\t - Websocket Port: " + str(channel.port) +"\n")
+    newSummary.append("\t - Websocket Url: " + str(channel.url) + "\n")
+    newSummary.append("\t - Messages Opcodes: \n")
+    newSummary.append("\t - Outgoing Messages format: \n")
+    newSummary.append("\t - Incoming Messages format: \n")
+    newSummary.append("\t - Input Messages encryption: \n")
+    newSummary.append("\t - Output Messages encryption: \n")
+    newSummary.append("\t - Ping/Pong: \n")
+    newSummary.append("\t - Close message: Opcode= , Text= \n")
+    newChannelElement.append_clean(newSummary);
+
+  print(handshakeMessage.getRequestHeader().getHostName())
```

Comment:
Remove?

Reply:
I'am not sure about. I think it is better now, make clear that something happened

Reply:
Makes sense, my initial impression was that this was a debug statement, worth adding "Processing [something]" (better to use the channel ID? it's guaranteed to exist) to be more obvious that's reporting progress, also worth adding a message at end to let know it finished.